app/assets/javascripts/thyme/metadata_manager.js

```diff
@@ -0,0 +1,221 @@
+          "class": 'material-icons'
+        });
+        const $scriptRef = $('<a/>', {
+          href: meta.script,
+          target: '_blank'
+        });
+        $scriptRef.append($scriptIcon);
+        if (meta.script === null) {
+          $scriptRef.hide();
+        }
+        const $quizIcon = $('<i/>', {
+          text: 'videogame_asset',
+          "class": 'material-icons'
+        });
+        const $quizRef = $('<a/>', {
+          href: meta.quiz,
+          target: '_blank'
+        });
+        $quizRef.append($quizIcon);
+        if (meta.quiz === null) {
+          $quizRef.hide();
+        }
+        const $extIcon = $('<i/>', {
+          text: 'link',
+          "class": 'material-icons'
+        });
+        const $extRef = $('<a/>', {
+          href: meta.link,
+          target: '_blank'
+        });
+        $extRef.append($extIcon);
+        if (meta.link === null) {
+          $extRef.hide();
+        }
+        const $description = $('<div/>', {
+          text: meta.text,
+          "class": 'mx-3'
+        });
+        const $explanation = $('<div/>', {
+          text: meta.explanation,
+          "class": 'm-3'
+        });
+        const $details = $('<div/>');
+        $details.append($link).append($description).append($explanation);
+        let $icons = $('<div/>', {
+          style: 'flex-shrink: 3; display: flex; flex-direction: column;'
+        });
+        $icons.append($videoRef).append($manRef).append($scriptRef).append($quizRef).append($extRef);
+        $listItem.append($details).append($icons);
+        $metaList.append($listItem);
+        $videoRef.on('click', function() {
+          video.pause();
+        });
+        $manRef.on('click', function() {
+          video.pause();
+        });
+        $extRef.on('click', function() {
+          video.pause();
+        });
+        $link.on('click', function() {
+          //displayBackButton();
+          video.currentTime = this.id.replace('l-', '');
+        });
+        let metaElement = $listItem.get(0);
+        thymeUtility.renderLatex(metaElement);
+      }
+      // store metadata start times as data attribute
+      $metaList.get(0).dataset.times = JSON.stringify(times);
+      // if user jumps to a new position in the video, display all metadata
+      // that start before this time and hide all that start later
+      $(video).on('seeked', function() {
+        const time = video.currentTime;
+        metadataManager.metaIntoView(time);
+      });
+      // if the metadata cue changes, highlight all current media and scroll
+      // them into view
+      $(metaTrack).on('cuechange', function() {
+        let j = 0;
+        const time = video.currentTime;
+        $('#' + metadataListId + ' li').removeClass('current');
+        while (j < this.activeCues.length) {
+          const activeStart = this.activeCues[j].startTime;
+          let metalink;
+          if (metalink = document.getElementById('m-' + activeStart)) {
```

Comment:
Prefer to assign data to the variable prior to the `if` statement. Then only do: `if (metalink)`